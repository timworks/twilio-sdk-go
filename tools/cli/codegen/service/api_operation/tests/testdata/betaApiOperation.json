{
  "packageName": "entity",
  "name": "Fetch",
  "imports": ["github.com/timworks/twilio-sdk-go/client"],
  "http": {
    "uri": "/Services/{serviceSid}/Entities/{identity}",
    "method": "Get",
    "pathParams": [
      {
        "name": "serviceSid",
        "value": {
          "onService": true,
          "type": "string",
          "property": "serviceSid"
        }
      },
      {
        "name": "identity",
        "value": {
          "onService": true,
          "type": "string",
          "property": "identity"
        }
      }
    ]
  },
  "documentation": {
    "description": "retrieves an entity resource",
    "twilioDocsLink": "https://www.twilio.com/docs/verify/api/entity#fetch-an-entity-resource"
  },
  "config": {
    "beta": true
  },
  "response": {
    "name": "FetchEntityResponse",
    "type": "JSON",
    "properties": [
      {
        "name": "AccountSid",
        "type": "string",
        "value": "account_sid",
        "required": true
      },
      {
        "name": "DateCreated",
        "type": "time.Time",
        "value": "date_created",
        "required": true
      },
      {
        "name": "DateUpdated",
        "type": "time.Time",
        "value": "date_updated",
        "required": false
      },
      {
        "name": "Identity",
        "type": "string",
        "value": "identity",
        "required": true
      },
      {
        "name": "ServiceSid",
        "type": "string",
        "value": "service_sid",
        "required": true
      },
      {
        "name": "Sid",
        "type": "string",
        "value": "sid",
        "required": true
      },
      {
        "name": "URL",
        "type": "string",
        "value": "url",
        "required": true
      }
    ],
    "documentation": {
      "description": "defines the response fields for the retrieved entity"
    }
  }
}
