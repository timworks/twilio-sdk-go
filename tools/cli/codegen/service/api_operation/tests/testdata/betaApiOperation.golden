// Package entity contains auto-generated files. DO NOT MODIFY
package entity

import (
	"context"
	"net/http"
	"time"

	"github.com/timworks/twilio-sdk-go/client"
)

// FetchEntityResponse defines the response fields for the retrieved entity
type FetchEntityResponse struct {
	AccountSid  string     `json:"account_sid"`
	DateCreated time.Time  `json:"date_created"`
	DateUpdated *time.Time `json:"date_updated,omitempty"`
	Identity    string     `json:"identity"`
	ServiceSid  string     `json:"service_sid"`
	Sid         string     `json:"sid"`
	URL         string     `json:"url"`
}

// Fetch retrieves an entity resource
// See https://www.twilio.com/docs/verify/api/entity#fetch-an-entity-resource for more details
// Context is defaulted to Background. See https://golang.org/pkg/context/#Background for more information
// This resource is currently in beta and subject to change. Please use with caution
func (c Client) Fetch() (*FetchEntityResponse, error) {
	return c.FetchWithContext(context.Background())
}

// FetchWithContext retrieves an entity resource
// See https://www.twilio.com/docs/verify/api/entity#fetch-an-entity-resource for more details
// This resource is currently in beta and subject to change. Please use with caution
func (c Client) FetchWithContext(context context.Context) (*FetchEntityResponse, error) {
	op := client.Operation{
		Method: http.MethodGet,
		URI:    "/Services/{serviceSid}/Entities/{identity}",
		PathParams: map[string]string{
			"serviceSid": c.serviceSid,
			"identity":   c.identity,
		},
	}

	response := &FetchEntityResponse{}
	if err := c.client.Send(context, op, nil, response); err != nil {
		return nil, err
	}
	return response, nil
}
